octree_web_viewer/src/bin/points_web_viewer.rs

```diff
@@ -12,44 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use octree_web_viewer::utils;
-use point_viewer::octree;
-use std::path::PathBuf;
+use octree_web_viewer::backend_error::PointsViewerError;
+use octree_web_viewer::state::AppState;
+use octree_web_viewer::utils::start_octree_server;
+use std::path::{Path, PathBuf};
 use std::sync::Arc;
 use structopt::StructOpt;
 
 /// HTTP web viewer for 3d points stored in OnDiskOctrees
 #[derive(StructOpt, Debug)]
 #[structopt(name = "points_web_viewer", about = "Visualizing points")]
-struct CommandlineArguments {
+pub struct CommandLineArguments {
     /// The octree directory to serve, including a trailing slash.
-    octree_path: String,
+    #[structopt(name = "DIR", parse(from_os_str))]
+    octree_path: PathBuf,
     /// Port to listen on.
-    #[structopt(default_value = "5433", long = "--port")]
+    #[structopt(default_value = "5433", long = "port")]
     port: u16,
     /// IP string.
-    #[structopt(default_value = "127.0.0.1", long = "--ip")]
+    #[structopt(default_value = "127.0.0.1", long = "ip")]
     ip: String,
+    #[structopt(default_value = "20", long = "cache_items")]
+    cache_max: usize,
+}
+
+/// init app state with command arguments
+/// backward compatibilty is ensured
+pub fn state_from(args: CommandLineArguments) -> Result<AppState, PointsViewerError> {
+    // initial implementation: suffix from args not yet supported
+    let suffix = PathBuf::new();
+    let prefix = args.octree_path.parent().unwrap_or_else(|| Path::new(""));
```

Comment:
Would `unwrap_or_default()` instead also work?

Reply:
it does not compile.

```
 note: the method `unwrap_or_default` exists but the following trait bounds were not satisfied:
           `&std::path::Path : std::default::Default`
```